source/draw/DrawState.ooc

```diff
@@ -132,9 +132,29 @@ DrawState: cover {
 	}
 	// Normalized transform
 	getTransformNormalized: func -> FloatTransform3D { this _transformNormalized }
+	// Reference point
+	setOrigin: func (origin: FloatPoint2D) -> This {
+		this _originReference = origin
+		this
+	}
+	// Local point
+	getOriginLocal: func -> FloatPoint2D {
+		version(safe)
+			raise(this target == null, "Can't get local origin relative to a target that does not exist.")
+		this _originReference + ((this target size toFloatVector2D()) / 2.0f)
```

Comment:
We need them where coordinate system is implicit and the rest is for consistent style.